src/Request.php

```diff
@@ -30,9 +30,25 @@ class Request
      */
     public function __construct()
     {
-        $this->requestData = $_REQUEST;
-        $this->postData = $_POST;
-        $this->getData = $_GET;
+        $this->setData();
+    }
+
+    /**
+     * Set requestData, postData, getData.
+     *
+     * @return null
+     */
+    public function setData()
```

Comment:
or you can substitute it with Laravel `request` class using service provider.

Reply:
As Multipay/Request use in verify method of drivers we can't use dependency injection, on other hand if we take it out of verify body and put in signature then shetabit/multipay wont work anymore because we don't have dependency injection mechanism in pure php. 
If I have any miss understanding here please led.

Reply:
in PHP we have dependency injection.
our target is Laravel in this issue. you can create another Request class and substitute it with the current one using Laravel's service provider.

Reply:
Then we should restructure driver's verify method signature, true?

Reply:
No. we don't need to touch drivers. I think you didn't understand service provider concept in this issue.

Reply:
@amirpourastarabadi We only have `config` function in Laravel so this method is not going to adapt with pure php

Reply:
@khanzadimahdi `Request` class is being used in drivers and we cannot inject an object to them because they are just calling a static method.
In the static call `Request` is instantiating itself and not using service container.  How is it possible to substitute the `Request` class while it's not being retrieved by the service container factory:
https://github.com/shetabit/multipay/blob/fb0e56932242638d316535ba854f2b7f77b01024/src/Drivers/Irankish/Irankish.php#L120
It could be possible if we had this:
`app(Request::class)->input('foo')`

Reply:
I will fix this one.

Reply:
thanks for your attention and dedication